### ISSUE TITLE
`citar-open-entry` is disabled in citar-embark

### ISSUE THREAD

**Issue by Elilif** [emacs-citar/citar#655]
**Describe the bug**
All functions are work well with the new package `citar-embark` except `citar-open-entry`, it doesn't read the entry where the point is located, and just message "Key ‘nil’ not found"

**To Reproduce**
1. Load this config
```
(use-package citar
  :custom
  (citar-bibliography your-bib-path))

(use-package citar-embark
  :ensure t
  :after citar embark
  :config (citar-embark-mode))
```
2. Call `citar-open` 
3. Select a entry and use `embark-act`, choose `citar-open-entry`
4. the minibuffer will message "Key ‘nil’ not found"

**Expected behavior**
`citar-open-entry` will jump to the correct bib entry

**Emacs version:"29.0.50"

**Comment by bdarcus** [comment]
Thanks.

We also need to fix this more generally; see #644.

**Comment by bdarcus** [comment]
> I'm afraid that it's different.

I know; I was just linking the issues.

But removing the interactive part won't work, since we also need it be interactive.

Hopefully @roshanshariff can take a quick look.
